__tests__/services/CleanupService.spec.ts

```diff
@@ -104,116 +150,644 @@ const mockCookie: CookiePropertiesCleanup = {
+        settings: {
+          ...firefoxState.settings,
+          debugMode: {
+            name: 'debugMode',
+            value: true,
+          },
+        },
+      }
+      beforeEach(() => {
+        when(global.browser.contextualIdentities.query)
+          .calledWith({})
+          .mockResolvedValue([{cookieStoreId: 'firefox-container-1'}] as never);
+        when(global.browser.cookies.getAllCookieStores)
+          .calledWith()
+          .mockResolvedValue([{id: 'firefox-default'}] as never);
+        when(global.browser.cookies.getAll)
+          .calledWith({storeId: 'firefox-default'})
+          .mockResolvedValue([
+            mockCookie, // no list
+            googleCookie, // greylist, opentab
+            youtubeCookie, // whitelist
+            yahooCookie,  // no list
+            openTabCookie, // opentab
+            githubCookie,  // greylist
+          ] as never);
+      });
+
+      it('Regular clean, exclude open tabs (Firefox).', async () => {
+        const ffResult = await cleanCookiesOperation(firefoxState, cleanupProperties);
+        expect(spyLib.isFirstPartyIsolate).toHaveBeenCalledTimes(1);
+        expect(global.browser.contextualIdentities.query).not.toHaveBeenCalled();
+        expect(global.browser.extension.isAllowedIncognitoAccess).toHaveBeenCalledTimes(1);
+        expect(global.browser.cookies.getAllCookieStores).toHaveBeenCalledTimes(1);
+        // getAll called from: isFirstPartyIsolate x1, directly with storeId: 'default' and 'firefox-default';
+        expect(global.browser.cookies.getAll).toHaveBeenCalledTimes(3);
+        expect(global.browser.cookies.remove).toHaveBeenCalledTimes(2);
+        expect(global.browser.browsingData.removeLocalStorage).not.toHaveBeenCalled();
+
+        expect(ffResult).not.toBe(undefined);
+        expect(ffResult!.cachedResults.dateTime.indexOf('12:34:56')).not.toBe(-1);
+        expect(ffResult!.cachedResults.recentlyCleaned).toBe(2);
+        expect(ffResult!.setOfDeletedDomainCookies.length).toBe(2);
+        expect(ffResult!.setOfDeletedDomainCookies).toEqual(['test.com','yahoo.com']);
+      });
+
+      it('If cleanupProperties is missing, presume Regular clean, exclude open tabs (Firefox).', async () => {
+        const ffResult = await cleanCookiesOperation(firefoxState);
+        expect(spyLib.isFirstPartyIsolate).toHaveBeenCalledTimes(1);
+        expect(global.browser.contextualIdentities.query).not.toHaveBeenCalled();
+        expect(global.browser.extension.isAllowedIncognitoAccess).toHaveBeenCalledTimes(1);
+        expect(global.browser.cookies.getAllCookieStores).toHaveBeenCalledTimes(1);
+        // getAll called from: isFirstPartyIsolate x1, directly with storeId: 'default' and 'firefox-default';
+        expect(global.browser.cookies.getAll).toHaveBeenCalledTimes(3);
+        expect(global.browser.cookies.remove).toHaveBeenCalledTimes(2);
+        expect(global.browser.browsingData.removeLocalStorage).not.toHaveBeenCalled();
+
```

Comment:
To be honest, if some or all of the `toHaveBeenCalled();` lines can be removed without decreasing coverage, then I rather not have it to not clutter the tests.

The only time it makes sense is that if you are testing something important like `        expect(global.browser.cookies.getAll).toHaveBeenCalledTimes(3);` where the input would affect this in the CleanupService.

Reply:
I'll try and minimize the amount of tests, but coverage requires all functions to be covered, irregardless of where it affects the extension...

Reply:
As the previous test have already covered it, I'll remove redundancy ones.

Reply:
Should be resolved through 2b5e3b2